### ISSUE TITLE
load_model fails with error ValueError: axes don't match array

### ISSUE THREAD

**Issue by tonmoyborah** [keras-team/keras#13392]
The discussion in #9562  moved towards multi-gpu but the problem persists in even single gpu setups. I have written a dummy script to reproduce the error as @Dref360 asked. @wt-huang I don't think the issue is all about 'multi-gpu', sure some had faced it because of that.
First the error doesn't show up if we use only basic layers. The InceptionResNetV2 brings out the issue. 
Secondly, the load_model takes an insanely long duration to execute.
Thirdly, if I freeze the layers as with Trainable = False before saving the model as someone in SO suggested, I instead get this warning "Error in loading the saved optimizer state. As a result, your model is starting with a freshly initialized optimizer" which I don't want.
I am opening this as a new issue instead of a comment as this could be a separate thing.
Please advise if otherwise.
The code:

```
import numpy as np
import os
from keras.layers import Input, Flatten, Dense, Dropout, Lambda, Conv2D, MaxPooling2D, Concatenate
from keras.models import Model, load_model
from keras import backend as K
from keras.optimizers import RMSprop
from keras.losses import cosine_proximity
from keras.applications.inception_resnet_v2 import InceptionResNetV2

os.environ['CUDA_VISIBLE_DEVICES'] = '0'

def euclidean_distance(vects):
	x, y = vects
	sum_square = K.sum(K.square(x - y), axis=1, keepdims=True)
	return K.sqrt(K.maximum(sum_square, K.epsilon()))


def eucl_dist_output_shape(shapes):
	shape1, shape2 = shapes
	return (shape1[0], 1)


def dummy_train_case():
	tr_pairs = np.array(np.zeros((10, 2, 128, 128, 3)))
	tr_y = np.array(np.zeros((10,)))
	te_pairs = np.array(np.zeros((11, 2, 128, 128, 3)))
	te_y = np.array(np.zeros((11,)))
	input_shape = tr_pairs.shape[2:]

	input = Input(shape=input_shape)
	base_model =InceptionResNetV2(include_top=False,weights=None,input_shape=input_shape)
	x = base_model(input)
	x = Flatten()(x)
	x = Dense(1024, activation='relu')(x)
	x = Dense(128, activation='relu')(x)
	base_network = Model(input, x)

	input_a = Input(shape=input_shape)
	input_b = Input(shape=input_shape)

	processed_a = base_network(input_a)
	processed_b = base_network(input_b)
	distance = Lambda(euclidean_distance, output_shape=eucl_dist_output_shape)([processed_a, processed_b])
	model = Model([input_a, input_b], distance)
	rms = RMSprop(lr=0.001)
	model.compile(loss=cosine_proximity, optimizer=rms)

	model.fit([tr_pairs[:, 0], tr_pairs[:, 1]], tr_y,
			  batch_size=48,
			  epochs=10,
			  validation_data=([te_pairs[:, 0], te_pairs[:, 1]], te_y))
	model.save(model_save_path)


if __name__ == '__main__':
	model_save_path = '/path/to/models/siamese_models/1.hf5'
	dummy_test_case()
	print('model saved')
	model = load_model(model_save_path)
	print('model loaded. Summary: {}'.format(model.summary()))
```
The full error stack:
Traceback (most recent call last):
File "/home/walle/PycharmProjects/keras_siamese/tests.py", line 60, in
model = load_model(model_save_path)
File "/home/walle/virtual_environments/tf1.13/lib/python3.5/site-packages/keras/engine/saving.py", line 419, in load_model
model = _deserialize_model(f, custom_objects, compile)
File "/home/walle/virtual_environments/tf1.13/lib/python3.5/site-packages/keras/engine/saving.py", line 274, in _deserialize_model
reshape=False)
File "/home/walle/virtual_environments/tf1.13/lib/python3.5/site-packages/keras/engine/saving.py", line 682, in preprocess_weights_for_loading
weights = convert_nested_model(weights)
File "/home/walle/virtual_environments/tf1.13/lib/python3.5/site-packages/keras/engine/saving.py", line 670, in convert_nested_model
original_backend=original_backend))
File "/home/walle/virtual_environments/tf1.13/lib/python3.5/site-packages/keras/engine/saving.py", line 682, in preprocess_weights_for_loading
weights = convert_nested_model(weights)
File "/home/walle/virtual_environments/tf1.13/lib/python3.5/site-packages/keras/engine/saving.py", line 658, in convert_nested_model
original_backend=original_backend))
File "/home/walle/virtual_environments/tf1.13/lib/python3.5/site-packages/keras/engine/saving.py", line 801, in preprocess_weights_for_loading
weights[0] = np.transpose(weights[0], (3, 2, 0, 1))
File "/home/walle/virtual_environments/tf1.13/lib/python3.5/site-packages/numpy/core/fromnumeric.py", line 639, in transpose
return _wrapfunc(a, 'transpose', axes)
File "/home/walle/virtual_environments/tf1.13/lib/python3.5/site-packages/numpy/core/fromnumeric.py", line 56, in _wrapfunc
return getattr(obj, method)(*args, **kwds)
ValueError: axes don't match array

**Comment by fengwang** [comment]
Same problem here. And I have trained my model for more than 1 month, then found I cannot load it........

**Comment by BigSeanZzz** [comment]
+1
anyone know the final solution of this issue?

**Comment by tonmoyborah** [comment]
I am currently using this workaround as already stated above - ' freeze the layers as with Trainable = False before saving the model'. Not ideal, still waiting to hear from the team

**Comment by felmiran** [comment]
Still no answer???
I am having the same problem. In my case, my model has a lambda layer which distorts the original image. The original and the distorted images are fed to an inception network in parallel, and the output for each is used calculate a custom loss function.

**Comment by davystrong** [comment]
Why was this issue closed? I'm facing the same issue with TF 2.8. Retraining everything and saving with different parameters shouldn't be the recommended way of fixing this.
